### ISSUE TITLE
Rule proposal: `array-join-separator`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1246]
Require [`separator`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join#parameters) when calling [`Array#join()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join). Even mean to use `','` to join, `array.join(',')` is more clear.

## Fail

```js
const string = array.join();
```

## Pass

```js
const string = array.join(',');
```
```js
const string = array.join('|');
```

**Comment by sindresorhus** [comment]
This is now accepted.
